docs/guides/contributing/documentation.rst

```diff
@@ -0,0 +1,1158 @@
+  cases, it isn't. Prioritize accuracy first and accessibility a close second.
+- **Be careful using words that have a special meaning in the context of
+  EdgeDB.** In casual speech or writing, you might talk about a "set" of
+  something in a generic sense. Using the word this way in EdgeDB documentation
+  might easily be interpreted as a reference to EdgeDB's :ref:`sets
+  <ref_eql_sets>`. Avoid this kind of casual usage of key terms.
+
+
+Where to Find It
+================
+
+Most of our documentation (including this guide) lives in `the edgedb
+repository <https://github.com/edgedb/edgedb/>`_ in `the docs directory
+<https://github.com/edgedb/edgedb/tree/master/docs>`_.
+
+Documentation for some of our client libraries lives inside the client's repo.
+If you don't find it in the edgedb repo at ``docs/clients``, you'll probably
+find it alongside the client itself. These clients will also have documentation
+stubs inside the edgedb repository directing you to the documentation's
+location.
+
+The `EdgeDB tutorial <https://www.edgedb.com/tutorial>`_ is part of `our web
+site repository <https://github.com/edgedb/website>`_. You'll find it in `the
+tutorial directory <https://github.com/edgedb/website/tree/main/tutorial>`_.
+
+Finally, our book for beginners titled `Easy EdgeDB
+<https://www.edgedb.com/easy-edgedb>`_ lives in `its own repo
+<https://github.com/edgedb/easy-edgedb>`_.
+
+
+How to Build It
+===============
+
+edgedb/edgedb
+-------------
+
+The ``edgedb`` repository contains all of its documentation in the ``docs/``
+directory. Run ``make docs`` to build the documentation in the edgedb repo. The
+repository contains a ``Makefile`` for all of Sphinx's necessary build options.
+The documentation will be built to ``docs/build``.
+
+To run tests, first :ref:`build EdgeDB locally
+<ref_guide_contributing_code_build>`. Then run ``edb test -k doc``.
+
+Building the docs from this repo will not give you a high-fidelity
+representation of what users will see on the web site. For that, you may want
+to do a full documentation build.
+
+Full Documentation Build
+------------------------
+
+A full documentation build requires more setup, but it is the only way to see
+your documentation exactly as the user will see it. This is not required, but
+it can be useful to help us review and approve your request more quickly by
+avoiding mistakes that would be easier to spot when they are fully rendered.
+
+To build, clone `our website repository <https://github.com/edgedb/website>`_
+and `follow the installation instructions
+<https://github.com/edgedb/website#installation>`_. Then run ``yarn dev`` to
+start a development server which also triggers a build of all the
+documentation.
+
+.. note::
+
+    Although the development server runs a watch task, it does not update
+    documentation as part of this. When you make changes to documentation, you
+    will need to interrupt and restart the development server.
```

Comment:
This behaviour is a bug. `yarn dev` should watch for docs changes and re-build automatically without needing to restart (with the small caveat that we can't communicate this update to nextjs hot reload, so you do have to manually refresh the page in the browser).